### DISCUSSION TITLE
What is the session server in code-server?

### DISCUSSION THREAD

**Discussion by tobycm** [coder/code-server#6636] [Q&A] [answered]
When I start code-server there is a line `Session server listening on /vsus/.local/share/code-server/code-server-ipc.sock` in output

What does session server means and what does it do?
Thanks

**Comment by code-asher** [comment, marked as answer]
The session server handles opening files in the right code-server session via the external command line (so not the integrated terminal).

So for example say you have code-server opened in your browser and on a separate command line you ssh into the server where code-server is hosted and you run `code-server /some/file`, it will ask the session server to open that file and the session server will pick the right code-server based on what folder it is displaying.  If there is no code-server window open it will try to start a new code-server process instead, so keep that in mind.  You can prevent this by adding `--reuse-window`.

If you run this same command from the integrated terminal, it does not use the session server, it just always uses the current window.
